From 7032eb6b4dcf60dd4ef2cc1b05d486ec80970954 Mon Sep 17 00:00:00 2001
From: Omair Mohammed Abdullah <omair.m.abdullah@intel.com>
Date: Tue, 23 Dec 2014 15:52:02 +0530
Subject: [PATCH 09/15] ASoC: DAPM: add a helper function to get the widget
 pointer

Issue: GMINL-6292

This patch adds a helper function to get the widget pointer

Change-Id: I38976e4eca3bbb656bfb97cba6eb53d8096b20af
Signed-off-by: Omair Mohammed Abdullah <omair.m.abdullah@intel.com>
Signed-off-by: Fabien Marotte <fabien.marotte@intel.com>
---
 include/sound/soc-dapm.h                         | 3 +++
 sound/soc/intel/platform-libs/controls_v2_dpcm.c | 2 +-
 sound/soc/soc-dapm.c                             | 8 ++++++++
 3 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/include/sound/soc-dapm.h b/include/sound/soc-dapm.h
index 4f61c1e..98781c9 100644
--- a/include/sound/soc-dapm.h
+++ b/include/sound/soc-dapm.h
@@ -165,6 +165,9 @@ void dapm_mark_io_dirty(struct snd_soc_dapm_context *dapm);
 /* dapm path query */
 int snd_soc_dapm_dai_get_connected_widgets(struct snd_soc_dai *dai, int stream,
 	struct snd_soc_dapm_widget_list **list);
+struct snd_soc_dapm_widget *snd_soc_dapm_find_widget(
+			struct snd_soc_dapm_context *dapm, const char *pin,
+			bool search_other_contexts);
 
 struct snd_soc_codec *snd_soc_dapm_kcontrol_codec(struct snd_kcontrol *kcontrol);
 
diff --git a/sound/soc/intel/platform-libs/controls_v2_dpcm.c b/sound/soc/intel/platform-libs/controls_v2_dpcm.c
index b15d982..02f4cf9 100644
--- a/sound/soc/intel/platform-libs/controls_v2_dpcm.c
+++ b/sound/soc/intel/platform-libs/controls_v2_dpcm.c
@@ -532,7 +532,7 @@ static int sst_voice_mode_put(struct snd_kcontrol *kcontrol,
 	struct sst_data *sst = snd_soc_platform_get_drvdata(platform);
 	struct soc_enum *e = (void *)kcontrol->private_value;
 	struct snd_soc_dapm_widget *w;
-	unsigned int max = e->max - 1;
+	unsigned int max = e->items - 1;
 	unsigned int val, orig;
 	int i;
 
diff --git a/sound/soc/soc-dapm.c b/sound/soc/soc-dapm.c
index cf8788d..49cf65a 100644
--- a/sound/soc/soc-dapm.c
+++ b/sound/soc/soc-dapm.c
@@ -2357,6 +2357,14 @@ static struct snd_soc_dapm_widget *dapm_find_widget(
 	return NULL;
 }
 
+struct snd_soc_dapm_widget *snd_soc_dapm_find_widget(
+			struct snd_soc_dapm_context *dapm, const char *pin,
+			bool search_other_contexts)
+{
+	return dapm_find_widget(dapm, pin, search_other_contexts);
+}
+EXPORT_SYMBOL_GPL(snd_soc_dapm_find_widget);
+
 static int snd_soc_dapm_set_pin(struct snd_soc_dapm_context *dapm,
 				const char *pin, int status)
 {
-- 
1.9.1

